internal/gather/gather_bytes.go

```diff
@@ -167,29 +167,30 @@ func (b *bytesReadSeekCloser) ReadAt(bs []byte, off int64) (int, error) {
 
 	n := copy(bs, curSlice[offset:])
 
-	// first chunk written, move on to the next
-	sliceNdx++
+	// if we were able to consume the entire slice then move to the next slice
```

Comment:
I should add to the comment here to make an important behaviour slightly more evident here:

“if we have not read the full curSlice then later on we will simply return n, nil and not try to work on the next slice”.